### PR TITLE
tweak: add new action hooks and deprecated existing

### LINKED ISSUE #123: Pass along identical arguments in dt_push_post actions
The `dt_push_post` action can be fired by NetworkSiteConnection and WordPressExternalConnection but they do not pass along identical or the same number of arguments:

- NetworkSiteConnection
`do_action( 'dt_push_post', $new_post, $post_id, $args, $this );`
- WordPressExternalConnection
`do_action( 'dt_push_post', $response, $post_body, $type_url, $post_id, $args, $this );`

This makes it impractical to reliably use the hook and access the variables.

### PR DESCRIPTION
**Pull request by ravinderk**
### Description of the Change

As [discussed](https://github.com/10up/distributor/issues/123#issuecomment-1474454938), I deprecated `dt_push_post` and added two action hooks, `dt_push_external_post` and `dt_push_network_post`.

Closes #123

### How to test the Change

- A notice should log in `debug.log` when use `dt_push_post` action hook
-  `dt_push_external_post` and `dt_push_network_post` action hooks should not cause any issue when use

### Changelog Entry

> Added -  `dt_push_external_post` and `dt_push_network_post` action hooks.
> Deprecated - dt_push_post action hook

### Credits

Props @dkotter @peterwilsoncc @dhanendran

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by ravinderk** [comment]
While checking existing action/filters hooks, I notice the following things:
- We only document one filter/action hook. As a developer, I notice the following issues:
     - I saw that in a few hooks, we pass class objects, meaning object types often differ, which can cause a fatal error. We should mention it in the hooks doc.
     - We show only one location of the filter doc. I think if we reuse the same filter at multiple locations and then mention this, it will be helpful for the developer
        
        List of a few hooks:
         - dt_push_post_args
         - dt_pull_post_terms
- We should mention multiple value types in the filter doc's return type. I noticed that `dt_remote_get` filter hook returns `WP_Post `, `array`, and` boolean`

Let me know if you want me to take action on the same pull request.

cc: @peterwilsoncc @dkotter

File: includes/classes/ExternalConnections/WordPressExternalConnection.php [hunk 1]
```diff
@@ -706,7 +706,28 @@ public function push( $post, $args = array() ) {
 		);
 
 		// Action documented in includes/classes/InternalConnections/NetworkSiteConnection.php.
-		do_action( 'dt_push_post', $response, $post_body, $type_url, $post_id, $args, $this );
+		do_action_deprecated(
+			'dt_push_post',
+			array( $response, $post_body, $type_url, $post_id, $args, $this ),
+			'2.0.0',
+			'dt_push_network_post|dt_push_external_post',
+			esc_html__( 'The dt_push_post action has been deprecated. Please use dt_push_network_post or dt_push_external_post instead.', 'distributor' )
```
Comment (peterwilsoncc) [inline includes/classes/ExternalConnections/WordPressExternalConnection.php]:
```suggestion
			'dt_push_network_post|dt_push_external_post'
```

With the message the notice will double up the info. 

> Hook dt_push_post is <strong>deprecated</strong> since version 2.0.0! Use dt_push_network_post|dt_push_external_post instead. The dt_push_post action has been deprecated. Please use dt_push_network_post or dt_push_external_post instead.

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 2]
```diff
@@ -124,14 +124,35 @@ public function push( $post, $args = array() ) {
...
+			 * @param {array}      $args        The arguments passed into wp_insert_post.
+			 * @param {Connection} $this        The Distributor connection being pushed to.
 			 */
-			do_action( 'dt_push_post', $new_post_id, $post_id, $args, $this );
+			do_action_deprecated(
+				'dt_push_post',
+				array( $new_post_id, $post_id, $args, $this ),
+				'2.0.0',
+				'dt_push_network_post|dt_push_external_post',
+				esc_html__( 'The dt_push_post action has been deprecated. Please use dt_push_network_post or dt_push_external_post instead.', 'distributor' )
```
Comment (peterwilsoncc) [inline includes/classes/InternalConnections/NetworkSiteConnection.php]: As above.

```suggestion
				'dt_push_network_post|dt_push_external_post'
```

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 3]
```diff
@@ -206,7 +227,16 @@ public function push( $post, $args = array() ) {
 		}
 
 		/** This filter is documented in includes/classes/InternalConnections/NetworkSiteConnection.php */
-		do_action( 'dt_push_post', $new_post_id, $post_id, $args, $this );
+		do_action_deprecated(
+			'dt_push_post',
+			array( $new_post_id, $post_id, $args, $this ),
+			'2.0.0',
+			'dt_push_network_post|dt_push_external_post',
+			esc_html__( 'The dt_push_post action has been deprecated. Please use dt_push_network_post or dt_push_external_post instead.', 'distributor' )
```
Comment (peterwilsoncc) [inline includes/classes/InternalConnections/NetworkSiteConnection.php:234]: As above.

```suggestion
			'dt_push_network_post|dt_push_external_post'
```

File: tests/php/WordPressExternalConnectionTest.php [hunk 4]
```diff
@@ -129,7 +130,6 @@ public function test_push() {
 			'post_excerpt' => 'post excerpt',
 			'post_name'    => 'slug',
 			'post_status'  => 'publish',
-			'post_type'    => $post_type,
```
Comment (peterwilsoncc) [inline tests/php/WordPressExternalConnectionTest.php:132]: Why was this removed?

**Review by peterwilsoncc — COMMENTED** [review]
A few minor notes inline.

File: tests/php/WordPressExternalConnectionTest.php (hunk 4 above)
Comment (ravinderk) [reply tests/php/WordPressExternalConnectionTest.php:132]: This is a duplicate entry in the array. Please check the whole array value for more information.

**Comment by ravinderk** [comment]
@peterwilsoncc I merged your suggestion and replied to your question. This pull request is ready to review.

**Review by peterwilsoncc — APPROVED** [review]
Thanks, let's figure out the other hooks you mentioned on a follow up PR.
